### PR TITLE
Demo - Vite + Squint + Solid

### PR DESCRIPTION
**Pull request by brandonstubbs**
This is the ported vite solid template

@martinklepsch This is where things get interesting, so `esbuild` will emit `React` JSX. We will need to be able to emit other types of JSX depending on the project.

https://github.com/amoutonbrady/esbuild-plugin-solid
https://github.com/unjs/unbuild also seems to support solid
https://github.com/solidjs/vite-plugin-solid/blob/main/src/index.ts#L315-L328 the official plugin is using babel.

But this makes me worry, we are going to become the filter between all these other plugins. We will have to support them all. I wonder if there is a different approach?

### CODE REVIEW

**Comment by brandonstubbs** [comment]
https://esbuild.github.io/content-types/#using-jsx-without-react we could configure esbuild. But I still worry about being the filter. I suspect this is why the HMR is not working in any of the demo's

If we want to support the transformations we could take a `jsx` option to the plugin that could specify `solid` for example

But I guess we have to be the filter? 🙃

**Comment by brandonstubbs** [comment]
See resolution in https://github.com/squint-cljs/vite-plugin-squint/pull/13

**Comment by martinklepsch** [comment]
Should this still be in draft?

**Comment by brandonstubbs** [comment]
@martinklepsch yeah at the moment until we get #13 merged then I can fix these other PR's

**Comment by brandonstubbs** [comment]
HMR is not working, looking into it

**Comment by brandonstubbs** [comment]
HMR works now, it doesn't keep state during a module reload. But neither do the official templates

**Comment by brandonstubbs** [comment]
@martinklepsch this ones ready to go if you're happy 🙂

File: demo/react/src/App.cljs [hunk 1]
```diff
@@ -18,7 +18,7 @@
           [:h1 "Vite + Squint + React"]
           [:div {:className "card"}
            [:button {:onClick #(set-count (inc count))}
-            "count is " count]
+            "counttt is " count]
```
Comment (martinklepsch) [inline demo/react/src/App.cljs]: Should be just `count` I guess?

File: demo/solid/src/index.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export * from "./main.cljs"
```
Comment (martinklepsch) [inline demo/solid/src/index.js:1]: Have you looked into whether this index.js workaround is still necessary with the various improvements you've made to the plugin lately?

File: index.mjs [hunk 3]
```diff
@@ -18,12 +18,16 @@ export default function viteSquint(opts = {}) {
         return { code: compiled, map: null };
       }
     },
-    resolveId(id, imported, options) {
+    resolveId(id, importer, options) {
+      if (/\.cljs.jsx$/.test(id)) {
+        // other plugins might make us resolve ourselves
+        return path.resolve(dirname(importer), id);
+      }
```
Comment (martinklepsch) [inline index.mjs]: Can you expand on this? Like I'm not sure I get the comment 😇

**Comment by martinklepsch** [comment]
@brandonstubbs Awesome, added a few comments while reviewing the code, nothing that it inherently blocking so I'll let you address whatever you like & merge after. :+1:

File: demo/react/src/App.cljs (hunk 1 above)
Comment (brandonstubbs) [reply demo/react/src/App.cljs]: Thanks good catch

File: index.mjs (hunk 3 above)
Comment (brandonstubbs) [reply index.mjs]: I expanded on this. Please let me know if this is more clear

Comment (martinklepsch) [reply index.mjs]: Thanks that makes a lot of sense!

File: demo/solid/src/index.js (hunk 2 above)
Comment (brandonstubbs) [reply demo/solid/src/index.js:1]: It is still necessary at the moment, the only way we can remove this is by adding https://vitejs.dev/guide/api-plugin#transformindexhtml rewriting the `cljs` script tag

Comment (brandonstubbs) [reply demo/solid/src/index.js:1]: I will open an issue to track this

Comment (brandonstubbs) [reply demo/solid/src/index.js:1]: Issue created: #17

**Pull request merged**